astropy/table/column.py

```diff
@@ -301,6 +308,32 @@ def __reduce__(self):
 
         return reconstruct_func, reconstruct_func_args, state
 
+    def get_item(self, item):
```

Comment:
Depending on whether or not people like #4075, if we merge it it might obviate the need for these methods.  With #4075 I was able to almost completely eliminate overhead related to overriding `__getitem__` in `Column` (at least for the basic case--`MaskedColumn.__getitem__` is still slowish due to the complexities of masked arrays, though in my PR it still adds less overhead than it did before).

And of course there's some overhead related to performing additional checks in the `__getitem__` implementation, but with some cleverness that can almost be completely eliminated too.